dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/AbstractNodeVisitor.java

```diff
@@ -88,121 +92,125 @@ public R visitNullChk(NullChkNode n, P p) {
     }
 
     // Binary operations
+    public R visitBinaryOperation(BinaryOperationNode n, P p) {
+        return visitNode(n, p);
+    }
+
     @Override
     public R visitStringConcatenate(StringConcatenateNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitNumericalAddition(NumericalAdditionNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitNumericalSubtraction(NumericalSubtractionNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitNumericalMultiplication(NumericalMultiplicationNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitIntegerDivision(IntegerDivisionNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitFloatingDivision(FloatingDivisionNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitIntegerRemainder(IntegerRemainderNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitFloatingRemainder(FloatingRemainderNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitLeftShift(LeftShiftNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitSignedRightShift(SignedRightShiftNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitUnsignedRightShift(UnsignedRightShiftNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitBitwiseAnd(BitwiseAndNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitBitwiseOr(BitwiseOrNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     @Override
     public R visitBitwiseXor(BitwiseXorNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
     }
 
     // Comparison operations
     @Override
     public R visitLessThan(LessThanNode n, P p) {
-        return visitNode(n, p);
+        return visitBinaryOperation(n, p);
```

Comment:
Should we add a distinction between the binary operations and the comparison and conditional operations, as indicated by the comments? They have the same superclasses, but one might want to handle them separately.